website/server/libs/payments.js

```diff
@@ -44,6 +45,158 @@ function _dateDiff (earlyDate, lateDate) {
   return moment(lateDate).diff(earlyDate, 'months', true);
 }
 
+/**
+ * Add a subscription to members of a group
+ *
+ * @param  group  The Group Model that is subscribed to a group plan
+ *
+ * @return undefined
+ */
+api.addSubscriptionToGroupUsers = async function addSubscriptionToGroupUsers (group) {
+  let members;
+  if (group.type === 'guild') {
+    members = await User.find({guilds: group._id}).select('_id purchased').exec();
+  } else {
+    members = await User.find({'party._id': group._id}).select('_id purchased').exec();
+  }
+
+  let promises = members.map((member) => {
+    return this.addSubToGroupUser(member, group);
+  });
+
+  await Promise.all(promises);
+};
+
+/**
+ * Add a subscription to a new member of a group
+ *
+ * @param  member  The new member of the group
+ *
+ * @return undefined
+ */
+api.addSubToGroupUser = async function addSubToGroupUser (member, group) {
+  let customerIdsToIgnore = [this.constants.GROUP_PLAN_CUSTOMER_ID, this.constants.UNLIMITED_CUSTOMER_ID];
+
+  let data = {
+    user: {},
+    sub: {
+      key: 'group_plan_auto',
+    },
+    customerId: 'group-plan',
+    paymentMethod: 'Group Plan',
+    headers: {},
+  };
+
+  let plan = {
```

Comment:
Only five keys are assigned by default. I'll double check what is redundant though.